src/components/PageLayout/LoginForm/LoginForm.tsx

```diff
@@ -1,64 +1,68 @@
-import {FC, FormEvent, FormEventHandler, useEffect, useRef, useState} from 'react'
+import {FC} from 'react'
+import {SubmitHandler, useForm} from 'react-hook-form'
 
-import {Login} from '@/types/api/generated/user'
+import {FormInput} from '@/components/FormItems/FormInput/FormInput'
 import {AuthContainer} from '@/utils/AuthContainer'
 
 import styles from './LoginForm.module.scss'
 
+type LoginFormValues = {
+  email: string
+  password: string
+}
+
+const defaultValues: LoginFormValues = {
+  email: '',
+  password: '',
+}
+
 interface ILoginForm {
   closeOverlay: () => void
 }
 
 export const LoginForm: FC<ILoginForm> = ({closeOverlay}) => {
-  const [formData, setFormData] = useState<Login>({email: '', password: ''})
-  const emailRef = useRef<HTMLInputElement>(null)
   const {login} = AuthContainer.useContainer()
+  const {
+    handleSubmit,
+    control,
+    formState: {errors},
+  } = useForm<LoginFormValues>({defaultValues})
 
-  useEffect(() => {
-    emailRef.current?.focus()
-  }, [])
-
-  const handleChange = (e: FormEvent<HTMLInputElement>) => {
-    const {name, value} = e.currentTarget
-    setFormData((prevData) => ({...prevData, [name]: value}))
-  }
-
-  const handleLogin: FormEventHandler<HTMLFormElement> = async (e) => {
+  const onSubmit: SubmitHandler<LoginFormValues> = async (data) => {
     // ak to tu nie je, robi to weird veci- prida to email a password do URL
     // ako query parametre, refreshne stranku a vobec nestihne pustit nas kod
```

Comment:
toto uz predpokladam neplati (viazalo sa to na `preventDefault`), tak asi mozeme zmazat koment